Marlin/i2cEncoder.h

```diff
@@ -0,0 +1,254 @@
+/*
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#if defined(I2C_ENCODERS_ENABLED)
+
+#ifndef I2CENC_H
+#define I2CENC_H
+
+#include "macros.h"
+
+//=========== Advanced / Less-Common Encoder Configuration Settings ==========//
+//    (perhaps could go along with main configuration in configuration.h)     //
```

Comment:
Perhaps these should go into the advanced configuration along with the other defines...
(the things that are optional or likely to be altered, at least)